python/ray/scripts/scripts.py

```diff
@@ -402,22 +410,34 @@ def stop():
     # Note that raylet needs to exit before object store, otherwise
     # it cannot exit gracefully.
     processes_to_kill = [
-        "raylet",
-        "plasma_store_server",
-        "raylet_monitor",
-        "monitor.py",
-        "redis-server",
-        "default_worker.py",  # Python worker.
-        " ray_",  # Python worker.
-        "org.ray.runtime.runner.worker.DefaultWorker",  # Java worker.
-        "log_monitor.py",
-        "reporter.py",
-        "dashboard.py",
+        # The first element is to filter ps results by command name (only the
+        # executable name). The second element is to filter ps results by
+        # command with all its arguments. See STANDARD FORMAT SPECIFIERS
+        # section of http://man7.org/linux/man-pages/man1/ps.1.html about comm
+        # and args. The two levels of filters can help avoid killing non-ray
+        # processes.
+        ["raylet", None],
+        ["plasma_store_server", None],
+        ["raylet_monitor", None],
+        [None, "monitor.py"],
+        ["redis-server", None],
+        [None, "default_worker.py"],  # Python worker.
+        [" ray_", None],  # Python worker.
+        [None, "org.ray.runtime.runner.worker.DefaultWorker"],  # Java worker.
+        [None, "log_monitor.py"],
+        [None, "reporter.py"],
+        [None, "dashboard.py"],
```

Comment:
Nice fix. But let's open a separate PR for this fix?

Reply:
Submitted. https://github.com/ray-project/ray/pull/5508